x/concentrated-liquidity/bench_test.go

```diff
@@ -0,0 +1,235 @@
+		}
+
+		// Craete a pool
+		poolId, err := s.App.PoolManagerKeeper.CreatePool(s.Ctx, clmodel.NewMsgCreateConcentratedPool(s.TestAccs[0], denom0, denom1, tickSpacing, sdk.MustNewDecFromStr("0.001")))
+		noError(err)
+
+		clKeeper := s.App.ConcentratedLiquidityKeeper
+
+		// Create first position to set a price of 1 and tick of zero.
+		tokenDesired0 := sdk.NewCoin(denom0, sdk.NewInt(100))
+		tokenDesired1 := sdk.NewCoin(denom1, sdk.NewInt(100))
+		tokensDesired := sdk.NewCoins(tokenDesired0, tokenDesired1)
+		_, _, _, _, _, err = clKeeper.CreatePosition(s.Ctx, poolId, s.TestAccs[0], tokensDesired, sdk.ZeroInt(), sdk.ZeroInt(), types.MinTick, types.MaxTick)
+
+		pool, err := clKeeper.GetPoolById(s.Ctx, poolId)
+		noError(err)
+
+		// Zero by default, can configure by setting a specific position.
+		currentTick := pool.GetCurrentTick()
+
+		// Setup numberOfPositions positions at random ranges
+		for i := 0; i < numberOfPositions; i++ {
+
+			var (
+				lowerTick int64
+				upperTick int64
+			)
+
+			if denomIn == denom0 {
+				// Decreasing price so want to be below current tick
+
+				// minTick <= lowerTick <= currentTick
+				lowerTick = rand.Int63n(currentTick.Int64()-types.MinTick+1) + types.MinTick
+				// Normalize lowerTick to be a multiple of tickSpacing
+				lowerTick = lowerTick + (tickSpacing - lowerTick%tickSpacing)
+				// lowerTick <= upperTick <= currentTick
+				upperTick = currentTick.Int64() - rand.Int63n(int64(math.Abs(float64(currentTick.Int64()-lowerTick))))
+			} else {
+				// Increasing price so want to be above current tick
+
+				// currentTick <= lowerTick <= upperTick
+				lowerTick := rand.Int63n(types.MaxTick-currentTick.Int64()+1) + currentTick.Int64()
+				// Normalize lowerTick to be a multiple of tickSpacing
+				lowerTick = lowerTick + (tickSpacing - lowerTick%tickSpacing)
+				// lowerTick <= upperTick <= maxTick
+				upperTick = types.MaxTick - rand.Int63n(int64(math.Abs(float64(types.MaxTick-lowerTick))))
+			}
+			// Normalize upperTick to be a multiple of tickSpacing
+			upperTick = upperTick - upperTick%tickSpacing
+
+			tokenDesired0 := sdk.NewCoin(denom0, sdk.NewInt(rand.Int63n(maxAmountDeposited)))
+			tokenDesired1 := sdk.NewCoin(denom1, sdk.NewInt(rand.Int63n(maxAmountDeposited)))
+
+			accountIndex := rand.Intn(len(s.TestAccs))
+
+			s.createPosition(accountIndex, poolId, tokenDesired0, tokenDesired1, lowerTick, upperTick)
+		}
+
+		// Setup numberOfPositions full range positions for deeper liquidity.
+		if shouldCreateFullRangePositions {
+			for i := 0; i < numberOfPositions; i++ {
+				lowerTick := types.MinTick
+				upperTick := types.MaxTick
+
+				tokenDesired0 := sdk.NewCoin(denom0, sdk.NewInt(maxAmountDeposited).MulRaw(5))
+				tokenDesired1 := sdk.NewCoin(denom1, sdk.NewInt(maxAmountDeposited).MulRaw(5))
```

Comment:
I am confused on why we define a maxAmountDeposited as the maximum we can deposit per position, and then we create a single position with this var by multiplying it by five. Maybe we should redefine it?

Reply:
^upvote for this comment

Reply:
changed to `maxAmountDepositedFullRange`